src/drivers/virtio/virtqueue/mod.rs

```diff
@@ -704,13 +670,12 @@ impl Transfer {
 						send_buff,
 						recv_buff,
 						..
-					} = self
-						.transfer_tkn
-						.as_mut()
-						.unwrap()
-						.buff_tkn
-						.as_mut()
-						.unwrap();
+					} = unsafe {
+						Rc::get_mut_unchecked(self.transfer_tkn.as_mut().unwrap())
+							.buff_tkn
+							.as_mut()
+							.unwrap()
+					};
```

Comment:
`Rc::get_mut` does not work here, right? We can't justify this unsafe, can we? We should mark this as `FIXME`, stating that this should be reworked in the future. It is a step in the right direction, having these invariants spread around instead of collected in `Pinned` and collectively dismissed.